### DISCUSSION TITLE
What is the best way to use AnsiConsole in Unit Tests?

### DISCUSSION THREAD

**Discussion by buzzfrog** [spectreconsole/spectre.console#831] [Q&A] [answered]
Hi
I have some problems with the global AnsiConsole in my unit test. The problem is that the text from my previous test is still there when I do AnsiConsole.ExportText in my next test.

I want to have a clear (empty) AnsiConsole in each test.

I have tried to do AnsiConsole.Clear() but that throws an exception.

Have I missed something? :)

**Comment by patriksvensson** [comment, marked as answer]
@buzzfrog Take a look at `TestConsole` (which implements `IAnsiConsole`) in the `Spectre.Console.Testing` NuGet package. This is what Spectre.Console uses for it's own tests as well.

Example of usage: https://github.com/spectreconsole/spectre.console/blob/5b2b4375a45d060ab33236787213a8407a4d61ab/test/Spectre.Console.Tests/Unit/AnsiConsoleTests.cs#L10

**Reply by buzzfrog** [reply]
Thanks. I'll look into your suggestion.
